constraints/src/test/java/gov/nasa/jpl/aerie/constraints/tree/ASTTests.java

```diff
@@ -693,30 +694,38 @@ public void testNestedForEachActivityViolations() {
         List.of(
             new ActivityInstance(1, "TypeA", Map.of(), Interval.between(4, 6, SECONDS)),
             new ActivityInstance(2, "TypeB", Map.of(), Interval.between(5, 7, SECONDS)),
-            new  ActivityInstance(3, "TypeA", Map.of(), Interval.between(9, 10, SECONDS))
+            new ActivityInstance(3, "TypeA", Map.of(), Interval.between(9, 10, SECONDS))
         ),
         Map.of(),
         Map.of()
     );
 
-    final var violation = new Violation("Constraint Test", 1L, ConstraintType.model, List.of(), List.of(), new Windows(interval(4, 6, SECONDS), true));
     final var result = new ForEachActivityViolations(
         "TypeA",
         "act",
         new ForEachActivityViolations(
             "TypeB",
             "act",
-            new Supplier<>(List.of(violation))
+            new UniqueSupplier<>(
+                () -> new ConstraintResult(
+                    List.of(new Violation(List.of(Interval.between(1, 2, SECONDS)), new ArrayList<>())),
+                    new ArrayList<>()
+                )
+            )
```

Comment:
What is this constraint? One violation for every A within 2 seconds of a B?

Reply:
This is not a real constraint. It just provides a dummy violation for each pair of activities A and B